config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -416,6 +440,432 @@ type MetricsServerConfig struct {
+	// or the string `automatic`, which indicates that the limit will be
+	// automatically calculated based on cluster capacity.
+	// To specify no limit, omit this field.
+	// The value must match the following pattern: ^(automatic|[0-9]+(B|KB|MB|GB|TB)?)$
+	// Minimum length is 1 character.
+	// Maximum length is 50 characters.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=50
+	// +optional
+	EnforcedBodySizeLimit string `json:"enforcedBodySizeLimit,omitempty"`
+	// externalLabels defines labels to be added to any time series or alerts when
+	// communicating with external systems such as federation, remote storage,
+	// and Alertmanager. By default, no labels are added.
+	// When omitted, no external labels are applied (default behavior).
+	// When specified, at least one label must be provided in the labels field.
+	// +optional
+	ExternalLabels ExternalLabels `json:"externalLabels,omitempty,omitzero"`
+	// logLevel defines the verbosity of logs emitted by Alertmanager.
+	// This field allows users to control the amount and severity of logs generated, which can be useful
+	// for debugging issues or reducing noise in production environments.
+	// Allowed values are Error, Warn, Info, and Debug.
+	// When set to Error, only errors will be logged.
+	// When set to Warn, both warnings and errors will be logged.
+	// When set to Info, general information, warnings, and errors will all be logged.
+	// When set to Debug, detailed debugging information will be logged.
+	// When omitted, this means no opinion and the platform is left to choose a reasonable default, that is subject to change over time.
+	// The current default value is `Info`.
+	// +optional
+	LogLevel LogLevel `json:"logLevel,omitempty"`
+	// nodeSelector defines the nodes on which the Pods are scheduled
+	// nodeSelector is optional.
+	//
+	// When omitted, this means the user has no opinion and the platform is left
+	// to choose reasonable defaults. These defaults are subject to change over time.
+	// The current default value is `kubernetes.io/os: linux`.
+	// +optional
+	// +kubebuilder:validation:MinProperties=1
+	// +kubebuilder:validation:MaxProperties=10
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+	// queryLogFile specifies the file to which PromQL queries are logged.
+	// This setting can be either a filename, in which
+	// case the queries are saved to an `emptyDir` volume
+	// at `/var/log/prometheus`, or a full path to a location where
+	// an `emptyDir` volume will be mounted and the queries saved.
+	// Writing to `/dev/stderr`, `/dev/stdout` or `/dev/null` is supported, but
+	// writing to any other `/dev/` path is not supported. Relative paths are
+	// also not supported.
+	// By default, PromQL queries are not logged.
+	// The value must be a valid filename.
+	// +optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=255
+	QueryLogFile string `json:"queryLogFile,omitempty"`
```

Comment:
There needs to be validations written for this field that enforces that valid path references are being provided by an end user